lib/quiz.js

```diff
@@ -6,8 +6,108 @@ var restrict_clause = require('./sql_restrictions.js')(),
     mysql = require('mysql');
 
 var library = {
-    "add": undefined,
+    "add":{
+        /*
+        +--------+------------+--------+---------------+--------+--------+------+--------+------------------+------+
+        | id     | time       | userid | ip            | course | module | cmid | action | url              | info |
+        +--------+------------+--------+---------------+--------+--------+------+--------+------------------+------+
+        | 855990 | 1355075808 |   1418 | 92.56.173.165 |    110 | quiz   | 7640 | add    | view.php?id=7640 | 29   |
+        +--------+------------+--------+---------------+--------+--------+------+--------+------------------+------+
+
+        url: id => mdl_course_modules.id
+
+        info => mdl_quiz.id
+        */
+        sql_old:
+            'SELECT log.*, ' +
+                'u.id AS u_id, u.username AS u_username, u.email AS u_email, ' +
+                'c.id AS course_id, c.shortname AS course_shortname, ' +
+                'cm.id AS cm_id, cm.module AS module_id, cm.section AS section, cm.instance AS quiz_id, ' +
+                'q.name AS quiz_name ' +
+                'FROM mdl_log log ' +
+                'JOIN mdl_course c ON log.course = c.id ' +
+                'LEFT JOIN mdl_course_modules cm ON log.cmid = cm.id ' +
+                'LEFT JOIN mdl_quiz q ON cm.instance = q.id ' +
+                'JOIN mdl_user u ON log.userid = u.id ' +
+                "WHERE log.module = 'quiz'" + 
+                "AND log.action = 'add'" + 
+                "AND " + restrict_clause,
+
+        sql_old_2pass: (row) => {
+            var m = row.url.match(/id=(\d+)/);
+            if (!m){
+                console.log('No course module ID in URL ' + JSON.stringify(row));
+                return null;
+            }
+        },
+
+        sql_match: (row) => {
+            return row.cm_id ?
+                mysql.format(
+                    'SELECT cm.id AS cm_id ' +
+                    'FROM mdl_course_modules cm ' +
+                    'LEFT JOIN mdl_course c ON c.id = cm.course ' + 
+                    'JOIN mdl_quiz q ON q.id = cm.instance ' + 
+                    'JOIN mdl_modules m ON m.id = cm.module ' + 
+                    'JOIN mdl_user u ON (BINARY u.email = ? OR u.username = ?) ' + 
+                    'WHERE m.name = ? AND c.shortname = ? AND q.name = ?',
+                    [
+                        row["u_email"],
+                        row["u_username"],
+                        'quiz',
+                        row["course_shortname"],
+                        row["quiz_name"],
+                    ]
+                )
+                :
```

Comment:
See comment from other PR - the ternary operator isn't being used correctly here.